### PR TITLE
add version.txt and github workflow to update file.

### LINKED ISSUE #207: Add a version number that is accessible within the manage_externals code base
### Summary of Issue:
Inside of the manage_externals structure, there is no apparent way to see what version of manage_externals is currently being used.  Either a flag --version to manage_externals or a file with the version information is requested.

### PR DESCRIPTION
**Pull request by jedwards4b**
add a version.txt file and a --version option to checkout_externals

Adds a pre-push hook to update the file version.txt when you push to the repository.   

User interface changes?: Yes 
Adds --version option 

Fixes: #207 

Testing:
  test removed:
  unit tests: All pass
  system tests: All pass
  manual testing: ./checkout_externals --version

### CODE REVIEW

**Comment by jedwards4b** [comment]
@erik - this uses a github action added to bumpversion.yml to write the version to a file version.txt on push to main. That file is read by ./checkout_externals --version

**Comment by jedwards4b** [comment]
I realized that I need to update the version.txt file before generating the tag but I need to get the new tag name before updating the file - this action does that.  Tested in https://github.com/jedwards4b/testrepo

File: version.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+manic-1.2.24-3-gba00e50
```
Comment (gold2718) [inline version.txt:1]: Is this going to get cleaned up when it is tagged?
I.s., will it be `manic-1.2.25` ?

**Review by gold2718 — APPROVED** [review]
I guess it is okay if it really works. It is a lot harder to try to figure out the new Microsoft TypeScript (:grimacing:) then the first (shell) version.

File: version.txt (hunk 1 above)
Comment (jedwards4b) [reply version.txt:1]: Yes it will be manic-1.2.25.

**Pull request merged**